File: python/test/history_api_test.py
import os
import sys
import unittest

sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), '../')))

from maplestory_openapi.api.kms import MapleStoryApi

api_key = os.environ['API_KEY']  # Your API Key
api = MapleStoryApi(api_key)


# 확률 정보 조회
class HistoryApi(unittest.IsolatedAsyncioTestCase):
    async def test_starforce_history(self):
        count = 1000
        response = await api.get_starforce_history(count)
        print(response)

    async def test_potential_history(self):
        count = 1000
        response = await api.get_potential_history(count)
        print(response)

    async def test_cube_history(self):
        count = 1000
        response = await api.get_cube_history(count)
        print(response)


if __name__ == '__main__':
    unittest.main()

File: js/test/historyApi.test.ts
import process from 'process';

import { toString } from './utils';
import { MapleStoryApi } from '../src/maplestory/api/kms/mapleStoryApi';

const apiKey = process.env.API_KEY!; // Your API Key
const api = new MapleStoryApi(apiKey);

describe('확률 정보 조회', () => {
  test('스타포스 강화 결과 조회', async () => {
    const count = 1000;
    const response = await api.getStarforceHistory(count);

    console.log(toString(response));
  });

  test('잠재능력 재설정 이용 결과 조회', async () => {
    const count = 1000;
    const response = await api.getPotentialHistory(count);

    console.log(toString(response));
  });

  test('큐브 사용 결과 조회', async () => {
    const count = 1000;
    const response = await api.getCubeHistory(count);

    console.log(toString(response));
  });
});
